nwb_conversion_tools/utils/si090nwbephyswriter.py

```diff
@@ -68,12 +68,11 @@ def _get_num_frames(self, segment_index=0):
 
     @default_return([])
     def _get_traces(self, channel_ids=None, start_frame=None, end_frame=None, return_scaled=True, segment_index=0):
-        if self.recording is not None:
-            return self.recording.get_traces(channel_ids=channel_ids,
-                                             start_frame=start_frame,
-                                             end_frame=end_frame,
-                                             return_scaled=return_scaled,
-                                             segment_index=segment_index)
+        return self.recording.get_traces(channel_ids=channel_ids,
```

Comment:
@Saksham20 I don't thin you have to decorate all functions..the `_get_traces` will never return None and so the `_get_times`, `_get_unit_spike_train` etc

The decorator IMO would be useful for 3-4 functions max, so I wonder whether we need it after all...

Reply:
the decorator also checks if we're calling some method/attribute on a NoneType (if self.recording/sorting etc are None). Then it also returns the default value also.

Reply:
I see. But if say `self.sorting` is None, then in my opinion the `_get_unit_spike_train` SHOULD raise an exception, as calling it would be a mistake. No?